src/js/plugins/plugin.search.js

```diff
@@ -0,0 +1,575 @@
+            </span>
+            Search
+          </span>
+          <div>
+            <form class="BRbooksearch mobile">
+              <input type="search" class="BRsearchInput" placeholder="Search inside"/>
+              <button type="submit" class="BRsearchSubmit"></button>
+            </form>
+            <div class="BRmobileSearchResultWrapper">Enter your search above.</div>
+          </div>
+        </li>`
+      ));
+    }
+    return $el;
+  };
+})(BookReader.prototype.buildMobileDrawerElement);
+
+/** @override */
+BookReader.prototype.buildToolbarElement = (function (super_) {
+  return function () {
+    const $el = super_.call(this);
+    if (this.enableSearch) {
+      const $BRtoolbarSectionSearch = $(
+        `<span class="BRtoolbarSection BRtoolbarSectionSearch">
+          <form class="BRbooksearch desktop">
+            <input type="search" class="BRsearchInput" val="" placeholder="Search inside"/>
+            <button type="submit" class="BRsearchSubmit">
+              <img src="${this.imagesBaseURL}icon_search_button.svg" />
+            </button>
+          </form>
+        </span>`
+      );
+      $BRtoolbarSectionSearch.insertAfter($el.find('.BRtoolbarSectionInfo'));
+    }
+    return $el;
+  };
+})(BookReader.prototype.buildToolbarElement);
+
+/** @override */
+BookReader.prototype.initToolbar = (function (super_) {
+  return function (mode, ui) {
+    super_.apply(this, arguments);
+
+    const onSubmitDesktop = (e) => {
+      e.preventDefault();
+      const val = $(e.target).find('.BRsearchInput').val();
+      if (!val.length) return false;
+      this.search(val);
+      return false;
+    };
+
+    const onSubmitMobile = (e) => {
+      e.preventDefault();
+      const val = $(e.target).find('.BRsearchInput').val();
+      if (!val.length) return false;
+      this.search(val, { disablePopup: true });
+      this.$('.BRmobileSearchResultWrapper').append(
+        `<div>Your search results will appear below.</div>
+          <div class="loader tc mt20"></div>`
+      );
+      return false;
+    };
+
+    // Bind search forms
+    this.$('.BRbooksearch.desktop').submit(onSubmitDesktop);
+    this.$('.BRbooksearch.mobile').submit(onSubmitMobile);
+
+    // Handle clearing the search results
+    this.$(".BRsearchInput").bind('input propertychange', () => {
+      if (this.value === "") this.removeSearchResults();
+    });
+  };
+})(BookReader.prototype.initToolbar);
+
+
+/**
+ * Submits search request
+ *
+ * @param { string } term
+ * @param { object } options
+ * @param { boolean } options.disablePopup
+ */
+BookReader.prototype.search = function(term, options) {
+  options = options !== undefined ? options : {};
+  const defaultOptions = {
+    /**
+     * @type { bool } - (default=false) goToFirstResult - jump to the first result
+     */
+    goToFirstResult: false,
+    /**
+     * @type { bool } - (default=false) disablePopup - don't show the modal progress
+     */
+    disablePopup: false
+  };
```

Comment:
Let's not remove the `error` and `success` options; those are still public options, so we don't know if someone who uses BR outside of ia is using them. We need to try to keep things as backwards compatible as possible, because we're not providing notice of breaking changes. This would require a v5 version bump.